features/login/impl/src/main/kotlin/io/element/android/features/login/impl/LoginFlowNode.kt

```diff
@@ -78,9 +85,26 @@ class LoginFlowNode @AssistedInject constructor(
 
     private val inputs: Inputs = inputs()
 
+    private var customChromeTabStarted = false
+
     override fun onBuilt() {
         super.onBuilt()
         defaultLoginUserStory.setLoginFlowIsDone(false)
+        lifecycle.subscribe(
+            onResume = {
+                if (customChromeTabStarted) {
+                    customChromeTabStarted = false
+                    // Workaround to detect that the Custom Chrome Tab has been closed
+                    // If there is no coming OidcAction (that would end this Node),
+                    // consider that the user has cancelled the login
+                    // by pressing back or by closing the Custom Chrome Tab.
+                    lifecycleScope.launch {
+                        delay(5000)
```

Comment:
Why the 5s delay here?

Reply:
This is sort of a compromise... If we wait more, the user will see a disabled button for a longer time.
If we wait less, we take the risk to cancel the login operation (the request to the homeserver with the Oidc token) before it finishes. The request does not take a long time, so maybe 2s is enough, but I did not want to take the risk.

The solution is not ideal, maybe we could try to detect that no OidcAction has been triggered (this should happen quite faster) and if this is the case we can cancel faster. I will think again about it.